### ISSUE TITLE
Products with no action items in CLC are not available in the PowerApps solutions

### ISSUE THREAD

**Issue by ladonnaq** [Azure/azure-sdk-tools#6638]
The Cosmos Postgres team is working on releasing SDKs for a post-GA product.  The product completed GA in 2019, so long before any of the process or Engagement tools/apps were around. 

- Service Tree link for the product that the team wants to onboard and associate with a Release Plan - https://microsoftservicetree.com/products/f2f0538e-b2b9-4976-8afa-d4715e525e65
- Service Tree link for the Service that is associated with the product - https://microsoftservicetree.com/services/461f15e6-e9f5-404e-b873-fcb306e8837d
- In the Release Planner app, if you set the context to the associated service, the SKU does not show up in the drop-down list.  The only product that shows up this one - https://microsoftservicetree.com/products/9eae4134-c080-476b-b56d-ba4d75ec2c4a but it is retired. 

We used the retired product and onboarded it so that way we could proceed with creating a release plan.  This was to unblock the service team.  However, this is not accurate as they are not doing work on a retired product.  

Note:  Cloud Lifecycle was not launched until April 2022.  Any service that had products that had not reached GA had CLC records created and action items were automatically completed.  For services with products that had completed GA, any CLC record will not be useful as most do not have any CLC action items.  They were signed off using the old form and CLC records were created without action items for most.

**Comment by JonathanCrd** [comment]
I took a look at this and found that the product *Hyperscale (Citus)* is not part of the `Product Profiles` table in PROD, which means that is it being ignored by the cloud flow that synchronize the products with Service Tree and the other one for the Service-Product mapping table.

The reason this is happening is because *Hyperscale (Citus)* does not have something called `360ExpandedActionItem`. We don't know what these Actions Items are about or how to check if in fact this product is missing them. I also wonder why the flow is filtering out these products in the first place. Anyway, I will keep looking at this and add a comment in this issue if I find more information about it.

In the meantime, @ladonnaq we have these questions:
- Do you know what these Actions Items are? Could you give us more context about it?
- Can you create Action Items for this product in service tree, so it doesn't get filtered out by our flow?

**Comment by maririos** [comment]
ping @ladonnaq in case you can help us here

**Comment by ladonnaq** [comment]
> I took a look at this and found that the product _Hyperscale (Citus)_ is not part of the `Product Profiles` table in PROD, which means that is it being ignored by the cloud flow that synchronize the products with Service Tree and the other one for the Service-Product mapping table.
> 
> The reason this is happening is because _Hyperscale (Citus)_ does not have something called `360ExpandedActionItem`. We don't know what these Actions Items are about or how to check if in fact this product is missing them. I also wonder why the flow is filtering out these products in the first place. Anyway, I will keep looking at this and add a comment in this issue if I find more information about it.
> 
> In the meantime, @ladonnaq we have these questions:
> 
> * Do you know what these Actions Items are? Could you give us more context about it?  **I have no clue but I can ask.  I will send an email to the S360 and CLC team to find out.** 
> * Can you create Action Items for this product in service tree, so it doesn't get filtered out by our flow?  **The action items in CLC should be our launch criteria.  I will ask the S360/CLC team and let you know.**

**Comment by maririos** [comment]
This is happening to another service-product (Azure Batch - Batch) :


I was looking more into it and found issue https://github.com/Azure/azure-sdk-tools/issues/5228 and look at Wei's comment: 
> There are no corresponding action items, while we sync products from CLC, if action items are missing, some fields will be empty, so we escape these products.
That's why they do not exist in Dev and Prod, which is correct.

For example:
- Service-Product that we have in our DB looks like this: https://vnext.s360.msftcloudes.com/cloudLifecycle/products?global=9:a5f80ee7-f138-4a2f-8fbf-14ad315e2877&blade=stage:4~kpiowner:All~Domains:All~program:All~_loc:cloudLifecycle
- Service/product with no action items:
  - Hyperscale (Citus): https://vnext.s360.msftcloudes.com/cloudLifecycle/products?global=9:f2f0538e-b2b9-4976-8afa-d4715e525e65&blade=stage:1~_loc:cloudLifecycle
  - Batch: https://vnext.s360.msftcloudes.com/cloudLifecycle/products?global=8:4ebe0f40-9ea7-45b0-bf47-c86cf700a073&blade=stage:4~_loc:cloudLifecycle

This means 2 things:
1. Our tooling depends on Service Tree and on CLC entries
2. Products need to have information entered in CLC

The question now is, why some products don't have that? @ladonnaq could you help us understand this? what is CLC used for at Microsoft?

**Comment by ladonnaq** [comment]
> This is happening to another service-product (Azure Batch - Batch) :
> 
> I was looking more into it and found issue #5228 and look at Wei's comment:
> 
> > There are no corresponding action items, while we sync products from CLC, if action items are missing, some fields will be empty, so we escape these products.
> > That's why they do not exist in Dev and Prod, which is correct.
> 
> For example:
> 
> * Service-Product that we have in our DB looks like this: https://vnext.s360.msftcloudes.com/cloudLifecycle/products?global=9:a5f80ee7-f138-4a2f-8fbf-14ad315e2877&blade=stage:4~kpiowner:All~Domains:All~program:All~_loc:cloudLifecycle
> * Service/product with no action items:
>   
>   * Hyperscale (Citus): https://vnext.s360.msftcloudes.com/cloudLifecycle/products?global=9:f2f0538e-b2b9-4976-8afa-d4715e525e65&blade=stage:1~_loc:cloudLifecycle
>   * Batch: https://vnext.s360.msftcloudes.com/cloudLifecycle/products?global=8:4ebe0f40-9ea7-45b0-bf47-c86cf700a073&blade=stage:4~_loc:cloudLifecycle
> 
> This means 2 things:
> 
> 1. Our tooling depends on Service Tree and on CLC entries
> 2. Products need to have information entered in CLC
> 
> The question now is, why some products don't have that? @ladonnaq could you help us understand this? what is CLC used for at Microsoft?

Background: CLC was launched in April 2022.  The CLC and APEX team spent most of 2022 onboarding teams to use CLC.  Through automation, CLC team migrated existing products, but I do not know if it was ALL products in service tree.  Because of this there could be products in CLC that do not have action items.  The action items showing or not showing is controlled by policies related to the S360 action item (APEX launch criteria).   
Initially the Shanghai team was using the CLC/S360 metadata to get the product mapping but products were missing (b/c they did not have CLC records).  When the issue was fixed, I thought the logic was changed to use service tree data to determine the product list for each service in the release planner.  Service tree is the source of truth for service product mapping and lifecycle metadata (launch dates, current lifecycle, ...).  CLC is using service tree metadata for the lifecycle dates, etc.  So if the service tree metadata is wrong, then CLC lifecycle dates are wrong. 

1.  Why are there dependencies on CLC to determine the service product mapping?   What is the effort to redesign and implement?  Service-product mapping should be determined using service tree not CLC/S360 data. 
2. Because CLC was not launched until April 2022, there will be gaps especially with older services/products.  The older products that were migrated to CLC have inconsisent data in regards to the CLC action items.  Some show action items and they were automatically marked complete, some show no action items at all, etc.

**Comment by maririos** [comment]
@ladonnaq  thank you so much for this context!!
To make sure I understood your recommendation, Eng team should look into 
>Service-product mapping should be determined using service tree not CLC/S360 data

Basically, remove the CLC dependency so teams that are non-APEX can use our tooling.

**Comment by maririos** [comment]
Moving out of blocked as it looks like we know the problem and it is on Eng to figure out how to remove the CLC dependency

**Comment by ladonnaq** [comment]
FYI - Hit this issue with another team last week.  This will be important because of TypeSpec so need to unblock the teams when we do so (January timeframe).   Thanks!
